trawl.el

```diff
@@ -164,6 +164,7 @@
     (let ((val (cdr (assq form trawl--variables))))
       (and val (trawl--peel val))))
    ((not (consp form)) form)
+   ((consp form) (list (car form) (cdr form)))
```

Comment:
Would you explain what this accomplishes? Do you have any sample code that crashes, or regexps that aren't discovered, without this change?

In any case, that line cannot be correct; it short-circuits out the rest of the cond-statement, and it doesn't recurse.

Reply:
I've scanned my collection of elisp files and for one of them (simple-wiki.el from emacswiki.org) I'm getting an error.

The version and url of simple-wiki.el :
```elisp
;; Version: 1.0.9
;; Keywords: hypermedia
;; URL: http://www.emacswiki.org/cgi-bin/wiki.pl?SimpleWikiEditMode
```

The error:
```
trawl--get-string-list: Wrong type argument: listp, "=+[ 	]*
"
```

The offending data are
``` elisp
(defvar simple-wiki-outline-patterns
  '("=+" . "=+[ \t]*\n")
  "Pair of patterns for `outline-regexp' and `outline-heading-end-regexp'.")
```
that's why I tried to add the consp case.

Reply:
Thanks for the example! This is tricky: trawl uses the heuristic that a -patterns name suffix implies a list of regexps, which has proven correct on all sources I've tested so far, but it doesn't account for improper lists, which this one is. I'll see what I can do about it.

Reply:
What about:
```elisp
((consp form) (cons (trawl--peel (car form)) (trawl--peel (cdr form))))
```